public/javascript/catbus.coffee

```diff
@@ -0,0 +1,107 @@
+#
+# @author Daryl Roberts
+# @license MIT
+#
+
+# This silly line is the require line and the jQuery function start
+define ['jquery'], ($) -> $(document).ready ->
+
+  # Cached jquery references
+  $body = $('body')
+
+  # @todo
+  # $dismissables = $('[js-catbus-dismiss]')
+  # closeForDismissableClick = ->
+  #   if $catbusTail.is(':visible') and catbusOpen
+  #     closeCatbus()
+
+  $catbus     = $('[js-catbus]')
+  $catbusTail = $('[js-catbus-tail]')
+  $clickables = $catbusTail.add('[js-catbus-button]')
+
+  # Local storage constants
+  LS_CATBUS_STATUS = 'catbus-status'
+  LS_STORED_PINNED = 'pinned'
+  LS_STORED_CLOSED = 'closed'
+
+  catbusOpen = $body.hasClass 'catbus-open'
+
+  ###*
+   * Tells us if local storage says the catbus is pinned open or not.
+  ###
+  catbusPinned = ->
+    status = localStorage.getItem LS_CATBUS_STATUS
+
+    pinned = false
+    pinned = true if status is LS_STORED_PINNED
+
+    return pinned
+
+  ###*
+   * Set the local storage parameter to open or closed.
+   *
+   * I'm doing this with local storage because if we use the disable animation
+   * trick from the demo, I Think we can cut out the need for anything to be
+   * done server side. That being said, I'm still using string values in case
+   * this needs to be switched back to cookies :)
+   *
+   * @param {boolean} open which way to set the value.
+  ###
```

Comment:
@ishotfirst What do you think about doing this with local storage instead of cookies.  Do you think we still want PHP to care if the catbus is open?